content/security.md

```diff
@@ -0,0 +1,380 @@
+});
+```
+
+#### Publications only re-run when the logged in user changes
+
+The data publications return will often be dependent on the currently logged in user, and perhaps some properties about that user - whether they are an admin, whether they own a certain document, etc.
+
+Publications are not reactive, and they only re-run when the currently logged in `userId` changes, which can be accessed through `this.userId`. Because of this, it's easy to accidentally write a publication that is secure when it first runs, but doesn't respond to changes in the app environment. Let's look at an example:
+
+```js
+// #1: Bad! If the owner of the list changes, the old owner will still see it
+Meteor.publish('list', function (listId) {
+  const list = Lists.findOne(listId);
+
+  if (! list.userId === this.userId) {
+    throw new Meteor.Error('list.unauthorized', 'This list doesn\'t belong to you.');
+  }
+
+  return Lists.find(listId, {
+    fields: {
+      name: 1,
+      incompleteCount: 1,
+      userId: 1
+    }
+  });
+});
+
+// #2: Good! When the owner of the list changes, the old owner won't see it anymore
+Meteor.publish('list', function (listId) {
+  return Lists.find({
+    _id: listId,
+    userId: this.userId
+  }, {
+    fields: {
+      name: 1,
+      incompleteCount: 1,
+      userId: 1
+    }
+  });
+});
+```
+
+In the first example, if the `userId` property on the selected list changes, the query in the publication will still return the data, since the security check in the beginning will not re-run. In the second example, we have fixed this by putting the security check in the returned query itself.
+
+Unfortunately, not all publications are as simple to secure as the example above. For more tips on how to use `reywood:publish-composite` to handle reactive changes in publications, see the data loading article.
+
+#### Passing options into publications
+
+For certain applications, for example pagination, you'll want to pass options into the publication to control things like how many documents should be sent to the client. There are some extra considerations to keep in mind for this particular case.
+
+1. **Passing a limit**: In the case where you are passing the `limit` option of the query from the client, make sure to set a maximum limit. Otherwise, a malicious client could request too many documents at once, which could raise performance issues.
+2. **Passing in a filter**: If you want to pass fields to filter on because you don't want all of the data, for example in the case of a search query, make sure to intersect the fields passed from the client with the fields it is allowed to see. Otherwise, a client could query on secret fields it's not supposed to be able to access.
+3. **Passing in fields**: If you want the client to be able to decide which fields of the collection should be fetched, make sure to intersect that with the fields that client is allowed to see, so that you don't accidentally send secret data to the client.
```

Comment:
Maybe as a summary, something like: "the basic story here is that you should be careful to only allow the inputs to make the publication *more* restricted, never less".